Sources/VLCPlaybackController.m

```diff
@@ -322,94 +322,13 @@ - (void)stopPlayback
 
 - (void)restoreAudioAndSubtitleTrack
 {
-    MLFile *item = [MLFile fileForURL:_mediaPlayer.media.url].firstObject;
+    VLCMLMedia *media = [_delegate mediaForPlayingMedia:_mediaPlayer.media];
 
-    if (item) {
-        _mediaPlayer.currentAudioTrackIndex = item.lastAudioTrack.intValue;
-        _mediaPlayer.currentVideoSubTitleIndex = item.lastSubtitleTrack.intValue;
+    if (media) {
+        _mediaPlayer.currentAudioTrackIndex = (int) media.audioTrackIndex;
+        _mediaPlayer.currentVideoSubTitleIndex = (int) media.subtitleTrackIndex + 1;
```

Comment:
for some reason directly storing and reaplying currentVideoSubTitleIndex is off by one